pkg/platform/platform.go

```diff
@@ -6,18 +8,25 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	// Architecture holds the runtime architecture of the process.
-	Architecture string
-	// OSType holds the runtime operating system type (Linux, …) of the process.
-	OSType string
-)
+// Architecture holds the runtime architecture of the process.
+//
+// Unlike [runtime.GOARCH] (which refers to the compiler platform),
+// Architecture refers to the running platform.
+//
+// For example, Architecture reports "x86_64" as architecture, even
+// when running a "linux/386" compiled binary on "linux/amd64" hardware.
+var Architecture string
+
+// OSType holds the runtime operating system type of the process. It is
+// an alias for [runtime.GOOS].
+//
+// Deprecated: use [runtime.GOOS] instead.
+const OSType = runtime.GOOS
 
 func init() {
 	var err error
 	Architecture, err = runtimeArchitecture()
 	if err != nil {
-		logrus.Errorf("Could not read system architecture info: %v", err)
+		logrus.WithError(err).Error("Could not read system architecture info")
```

Comment:
FWIW, I wonder if we should consider replacing `var Architecture` with a `func Architecture()` (with a `sync.Once`). Currently this is part of an `init()`, which means that we'll be loading `"kernel32.dll"` on Windows (and a syscall), and we may be logging some error just by importing this package 🤔